File: backend/WebApi.Domain/src/Entities/Product.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace WebApi.Domain.src.Entities
{
    public class Product: BaseEntity
    {
        public string? Title { get; set; }
        public List <string>? Images { get; set; }
        public string? Description { get; set; }
        public decimal Price { get; set; }   

        [ForeignKey(nameof(Category))]
        public Guid CategoryId { get; set; }
        public Category Category { get; set; }
        public List<CartItem> CartItems { get; set; }
        public List<OrderProduct> OrderProducts { get; set; }
        
    }
}

File: frontend/src/types/Product.ts
import { Category } from "./Category";


export interface Product {
    id: string,
    title: string,
    images: string,
    description: string,
    price: number,
    category: Category
}

export interface NewProduct {
    title: string,
    images: string,
    description: string,
    price: number,
    category: Category
}

export interface ProductUpdate {
    id: string
    update:
    {
        title: string,
        images: string,
        description: string,
        price: number,
    },
    category: Category
}